pkgs/by-name/e2/e2fsprogs/package.nix

```diff
@@ -37,30 +61,35 @@ stdenv.mkDerivation rec {
     pkg-config
     texinfo
   ];
-  buildInputs = [
-    libuuid
-    gettext
-  ] ++ lib.optionals withFuse [ fuse3 ];
+  buildInputs =
+    [
+      libuuid
+      gettext
+    ]
+    ++ lib.optionals withFuse [ fuse3 ]
+    ++ lib.optionals withLibarchive [ libarchive' ];
 
   configureFlags =
-    if stdenv.hostPlatform.isLinux then
-      [
-        # It seems that the e2fsprogs is one of the few packages that cannot be
-        # build with shared and static libs.
-        (if shared then "--enable-elf-shlibs" else "--disable-elf-shlibs")
-        "--enable-symlink-install"
-        "--enable-relative-symlinks"
-        "--with-crond-dir=no"
-        # fsck, libblkid, libuuid and uuidd are in util-linux-ng (the "libuuid" dependency)
-        "--disable-fsck"
-        "--disable-libblkid"
-        "--disable-libuuid"
-        "--disable-uuidd"
-      ]
-    else
-      [
-        "--enable-libuuid --disable-e2initrd-helper"
-      ];
+    lib.optionals stdenv.isLinux [
```

Comment:
This should stay as stdenv.hostPlatform.isLinux (and below). We had a treewide replacement in that direction recently.